src/cascadia/Remoting/WindowManager.cpp

```diff
@@ -138,6 +138,7 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
                 // commandline in an existing window, or a new one, but either way,
                 // this process doesn't need to make a new window.
 
+                LOG_IF_WIN32_BOOL_FALSE(AllowSetForegroundWindow(static_cast<DWORD>(_monarch.GetPID())));

```

Comment:
Ideally, it should go before `_monarch.ProposeCommandline(...)`:

https://github.com/microsoft/terminal/blob/9317d42045dbf44e3c8f5991acf9dd30a8fc5987/src/cascadia/Remoting/WindowManager.cpp#L262-L273

The try block is within a loop that retries connecting to a different monarch in case we fail to propose to the current one.